### PR TITLE
[ONNX][dynamo_export] Integrate onnx-rewriter optimizer

### PR DESCRIPTION
**Pull request by BowenBao**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #123379

Introduces common standard onnx optimization such as constant, if, controlflow folding and pattern rewrites.

### CODE REVIEW

File: torch/onnx/_internal/exporter.py [hunk 1]
```diff
@@ -1232,6 +1232,16 @@ def export(self) -> ONNXProgram:
...
             )
 
+            try:
+                from onnxscript import optimizer
+
+                onnx_model = optimizer.optimize(onnx_model)
+            except ImportError:
+                warnings.warn(
+                    "ONNXScript optimizer is not available. Skipping optimization. "
+                    "Please install latest onnxscript to enable post export optimization."
```
Comment (thiagocrepaldi) [inline torch/onnx/_internal/exporter.py:1242]:
```suggestion
                    "Please `pip install onnxscript` to enable post-export optimization."
```

**Review by thiagocrepaldi — APPROVED** [review]

File: torch/onnx/_internal/exporter.py (hunk 1 above)
Comment (thiagocrepaldi) [reply torch/onnx/_internal/exporter.py:1242]: giving the users a "copy&paste" solution :)

**Review by justinchuby — APPROVED** [review]

File: torch/onnx/_internal/onnxruntime.py [hunk 2]
```diff
@@ -922,6 +922,20 @@ def maybe_map_to_meta_val(value):
                 opset_version=self._resolved_onnx_exporter_options.onnx_registry.opset_version,
             )
 
+            try:
+                from onnxscript import optimizer  # type: ignore[import]
+                from onnxscript.rewriter import (  # type: ignore[import]
+                    onnxruntime as ort_rewriter,  # type: ignore[import]
```
Comment (titaiwangms) [inline torch/onnx/_internal/onnxruntime.py:928]: Why do we only have onnxruntime rewriter here?

**Pull request closed without merging**